lib/entry-index.js

```diff
@@ -1,20 +1,41 @@
 'use strict'
 
-const asyncMap = require('slide/lib/async-map')
+module.exports = {
+  find,
+  insert,
+  ls,
+  lsStream,
+  delete: del,
+  _hashKey: hashKey,
+  _bucketPath: bucketPath
+}
```

Comment:
My only concern about this is that it breaks from the style that both cacache and pacote have been written in. I think if we're gonna do this hoisting, we should do it everywhere. I don't really have any strong technical opinion about either way of doing things (exports-near-definitions is more what I've seen from ES6 modules, and it's the way the main npm codebase does exports, which is the whole reasoning why I did it this way).

So, I defer to you for the decision, with the constraint that I'd like it to be one or the other, not a mix throughout the codebase :)